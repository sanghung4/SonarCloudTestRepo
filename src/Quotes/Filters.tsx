import React, { ChangeEvent, FormEvent } from 'react';

import {
  Box,
  Grid,
  TextField,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import FilterActionButtons from 'common/TablePageLayout/FilterActionButtons';

type Props = {
  search: string;
  setSearch: (search: string) => void;
  onSubmit: () => void;
  onReset: () => void;
  resultsCount?: number;
};

function Filters(props: Props) {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  return (
    <form onSubmit={handleSubmit}>
      <Grid container spacing={2} alignItems="flex-end">
        <Grid item xs={isSmallScreen ? 12 : 4}>
          <TextField
            label={t('common.searchBy')}
            placeholder={t('quotes.searchPlaceholder')}
            value={props.search}
            onChange={(e: ChangeEvent<HTMLInputElement>) =>
              props.setSearch(e.target.value)
            }
            fullWidth
          />
        </Grid>
        <FilterActionButtons
          dirty={Boolean(props.search && props.search !== '')}
          hideApplyOnMobile={false}
          resultsCount={props.resultsCount}
          onViewResults={props.onSubmit}
          onReset={props.onReset}
        />
      </Grid>
      <Box component="button" type="submit" sx={{ display: 'none' }} />
    </form>
  );

  function handleSubmit(e: FormEvent<HTMLFormElement>) {
    e.preventDefault();

    props.onSubmit();
  }
}

export default Filters;
