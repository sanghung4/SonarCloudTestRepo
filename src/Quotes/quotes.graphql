query Quotes($accountId: ID!) {
  quotes(accountId: $accountId) {
    ...Quote
  }
}
