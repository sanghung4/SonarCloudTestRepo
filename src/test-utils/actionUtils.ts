import { fireEvent, findByTestId, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export const fillFormTextInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const fillMaskedFormTextInput = async (
  testId: string,
  value: string
) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await userEvent.type(textInput, value);
  });
};

export const clickButton = async (testId: string) => {
  await act(async () => {
    const button = await findByTestId(document.body, testId);
    await fireEvent.click(button);
  });
};

export const clickRadio = async (testId: string, optionIndex: number) => {
  await act(async () => {
    const radio = await findByTestId(
      document.body,
      `${testId}-radio-${optionIndex}`
    );
    await userEvent.click(radio);
  });
};

export const fillFormSelectInput = async (testId: string, value: string) => {
  await act(async () => {
    const selectInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(selectInput, { target: { value } });
    await fireEvent.keyDown(window, { key: 'Enter' });
  });
};

export const focusInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.focus(input);
  });
};

export const changeInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, testId);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const blurInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.blur(input);
  });
};

export const keyboardDown = async (
  key: string,
  element?: Element | Node | Document | Window
) => {
  await act(async () => {
    await fireEvent.keyDown(element ?? window, { key });
  });
};
