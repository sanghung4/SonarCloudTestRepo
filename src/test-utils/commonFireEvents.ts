import { act, fireEvent } from '@testing-library/react';

export function actClickOnElement(element: HTMLElement) {
  act(() => {
    fireEvent.click(element);
  });
}
export function actUpdateInput(input: HTMLElement, value: string) {
  act(() => updateInput(input, value));
}
export function updateInput(input: HTMLElement, value: string) {
  fireEvent.focus(input);
  fireEvent.change(input, { target: { value } });
  fireEvent.blur(input);
}
