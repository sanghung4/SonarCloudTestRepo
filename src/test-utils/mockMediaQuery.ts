import mediaQuery from 'css-mediaquery';

function createMatchMedia(width: number) {
  return (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: () => jest.fn(),
    removeListener: () => jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  });
}

export const breakpoints = {
  mobile: 414,
  desktop: 1200
};

export const setBreakpoint = (breakpoint: 'mobile' | 'desktop' = 'desktop') => {
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: breakpoints[breakpoint]
  });
  window.dispatchEvent(new Event('resize'));

  window.matchMedia = createMatchMedia(window.innerWidth);
};
