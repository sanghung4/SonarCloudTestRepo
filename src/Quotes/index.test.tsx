import React from 'react';

import { waitFor } from '@testing-library/react';
import Quotes from 'Quotes';
import { render } from 'test-utils/TestWrapper';
import { setBreakpoint } from 'test-utils/mockMediaQuery';

describe('Quotes test', () => {
  it('should match snapshot on desktop', async () => {
    setBreakpoint('desktop');
    const { container } = render(<Quotes />);

    // Wait for debounce to fire
    await waitFor(() => new Promise((res) => setTimeout(res, 350)));

    expect(container).toMatchSnapshot();
  });
  it('should match snapshot on mobile', async () => {
    setBreakpoint('mobile');
    const { container } = render(<Quotes />);

    // Wait for debounce to fire
    await waitFor(() => new Promise((res) => setTimeout(res, 350)));

    expect(container).toMatchSnapshot();
  });
});
