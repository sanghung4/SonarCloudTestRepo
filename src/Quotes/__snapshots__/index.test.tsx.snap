// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes test should match snapshot on desktop 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1kbjjd5-MuiContainer-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1yp6lty-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-n1sla7-MuiBreadcrumbs-separator"
        >
          <svg>
            chevron-right.svg
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
            data-testid="quotes-breadcrumb"
          >
            Quotes
          </span>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-1obe4va-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10 css-1s6wx5p-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
            data-testid="quotes-title"
          >
            Quotes
          </h5>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gj3ka-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-1wcaknn"
      >
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-jhdc2n-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1rqx6ho-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-1"
                  id="mui-1-label"
                >
                  Search by
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-tg1mzk-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1mjnin0-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-1"
                    placeholder="Quote #, PO #, Requested By"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1z7n62"
                    >
                      <span>
                        Search by
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-190vp1s-MuiButtonBase-root-MuiButton-root"
                data-testid="branch-filter-set-filter"
                disabled=""
                tabindex="-1"
                type="button"
              >
                View Results
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-xmhp4d-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-md-true css-1xqcy2u-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-g1pjat-MuiTypography-root"
                  data-testid="results"
                >
                  0 results
                </h6>
              </div>
            </div>
          </div>
          <button
            class="MuiBox-root css-1hyfx7x"
            type="submit"
          />
        </form>
      </div>
      <div
        class="MuiTableContainer-root css-8cb3ox-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-qzyhip-MuiTable-root"
          data-testid="quotes-table"
          role="table"
          style="min-width: 0px;"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-bqkitb-MuiTableRow-root"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Quote #
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Status
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Acct #
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 150 0 auto; min-width: 0px; width: 150px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Job Name
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      PO #
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Requested Date
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Expiration Date
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Quote Total
                    </div>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                colspan="6"
              >
                <div
                  class="MuiBox-root css-18glhic"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 css-nri8m1-MuiTypography-root"
                  >
                    No results were found.
                  </h4>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-11ezxrp"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1vbz8i-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-kqwn4m-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-19z5v8g"
              data-testid="total-quotes-count"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Quotes test should match snapshot on mobile 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-b628xg-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-3u2h7w"
      data-testid="back-button-mobile"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
        href="/"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-lw1dhn-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
          >
            <svg>
              chevron-left.svg
            </svg>
          </span>
          Back to Home
        </button>
      </a>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-15ug3si-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10 css-wd0fv5-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
            data-testid="quotes-title"
          >
            Quotes
          </h5>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gj3ka-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-19kzrtu"
      >
        <form>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-jhdc2n-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-2"
                  id="mui-2-label"
                >
                  Search by
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-tg1mzk-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1mjnin0-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-2"
                    placeholder="Quote #, PO #, Requested By"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1z7n62"
                    >
                      <span>
                        Search by
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-190vp1s-MuiButtonBase-root-MuiButton-root"
                data-testid="branch-filter-set-filter"
                disabled=""
                tabindex="-1"
                type="button"
              >
                View Results
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-xmhp4d-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-md-true css-wcbppb-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-g1pjat-MuiTypography-root"
                  data-testid="results"
                >
                  0 results
                </h6>
              </div>
            </div>
          </div>
          <button
            class="MuiBox-root css-1hyfx7x"
            type="submit"
          />
        </form>
      </div>
      <div
        class="MuiTableContainer-root css-8cb3ox-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-qzyhip-MuiTable-root"
          data-testid="quotes-table"
          role="table"
          style="min-width: 0px;"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-bqkitb-MuiTableRow-root"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 125 0 auto; min-width: 0px; width: 125px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Quote #
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Status
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 75 0 auto; min-width: 0px; width: 75px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Acct #
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 150 0 auto; min-width: 0px; width: 150px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Job Name
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      PO #
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Requested Date
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Expiration Date
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap css-129ezgh-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Quote Total
                    </div>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                colspan="6"
              >
                <div
                  class="MuiBox-root css-18glhic"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 css-nri8m1-MuiTypography-root"
                  >
                    No results were found.
                  </h4>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-11ezxrp"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column-reverse MuiGrid-grid-xs-12 css-ykx38k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-kqwn4m-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-19z5v8g"
              data-testid="total-quotes-count"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
